### PR TITLE
bot: change mergify config for k8s versions

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

Since master and release branches run a different version of Kubernetes,
the conditions must be adapted.

Closes: https://github.com/rook/rook/issues/8865
Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #$https://github.com/rook/rook/issues/8865

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Review by subhamkrai — APPROVED** [review]

**Review by subhamkrai — CHANGES_REQUESTED** [review]
Do we need the same for 1.6?

File: .mergify.yml [hunk 1]
```diff
@@ -121,11 +91,12 @@ pull_request_rules:
       - 'check-success=encryption-pvc-db-wal'
       - 'check-success=encryption-pvc-kms-vault-token-auth'
       - 'check-success=TestCephSmokeSuite (v1.16.15)'
```
Comment (subhamkrai) [inline .mergify.yml:93]:
```suggestion
      - 'check-success=TestCephSmokeSuite (v1.15.12)'
```

File: .mergify.yml [hunk 2]
```diff
@@ -121,11 +91,12 @@ pull_request_rules:
...
-      - 'check-success=TestCephSmokeSuite (v1.20.2)'
+      - 'check-success=TestCephSmokeSuite (v1.21.0)'
       - 'check-success=TestCephHelmSuite (v1.16.15)'
```
Comment (subhamkrai) [inline .mergify.yml:95]:
```suggestion
      - 'check-success=TestCephHelmSuite (v1.15.12)'
```

**Review by subhamkrai — CHANGES_REQUESTED** [review]
need to change these two also. right?

File: .mergify.yml (hunk 1 above)
Comment (leseb) [reply .mergify.yml:93]: No, that's how the CI is setup, so I'm not changing it.

File: .mergify.yml (hunk 2 above)
Comment (leseb) [reply .mergify.yml:95]: ditto

File: .mergify.yml (hunk 1 above)
Comment (subhamkrai) [reply .mergify.yml:93]: okay :+1:

**Review by subhamkrai — APPROVED** [review]

**Review by travisn — APPROVED** [review]

**Pull request merged**